File: api/project/project.go
package project

type Project struct {
	Id                   string            `json:"id" bson:"_id,omitempty"`
	Name                 string            `json:"name"`
	Description          string            `json:"description"`
	Tags                 []string          `json:"tags"`
	Owner                string            `json:"owner"`
	ImageProvider        string            `json:"imageProvider"`
	ImageProviderOptions map[string]string `json:"imageProviderOptions"`
}

File: ui/src/app/classes/project.ts
export class Project {
  id: string;
  name: string;
  banner: string;
  owner: string;
  description: string;
  tags: Array<String>;
  imageProvider: string;
  imageProviderOptions: any;

  constructor() {
    this.id = "";
    this.name = "";
    this.owner = "";
    this.description = "";
    this.tags = new Array<string>();
    this.imageProvider = "";
    this.imageProviderOptions = {};
  }
}

export class ProjectStat {
  project: Project
  annotations: number
  contributors: number
}
